DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -57,31 +96,47 @@ function Get-TargetResource
             $($LocalizedData.FirewallRuleDoesNotExistMessage) -f $Name
             ) -join '')
         return @{
-            Name   = $Name
             Ensure = 'Absent'
+            Name   = $Name
         }
     }
 
     $properties = Get-FirewallRuleProperty -FirewallRule $firewallRule
 
-    # Populate the properties for get target resource
-    return @{
-        Name            = $Name
-        Ensure          = 'Present'
-        DisplayName     = $firewallRule.DisplayName
-        Group           = $firewallRule.Group
-        DisplayGroup    = $firewallRule.DisplayGroup
-        Enabled         = $firewallRule.Enabled
-        Action          = $firewallRule.Action
-        Profile         = $firewallRule.Profile.ToString() -replace(' ', '') -split(',')
-        Direction       = $firewallRule.Direction
-        Description     = $firewallRule.Description
-        RemotePort      = @($properties.PortFilters.RemotePort)
-        LocalPort       = @($properties.PortFilters.LocalPort)
-        Protocol        = $properties.PortFilters.Protocol
-        Program         = $properties.ApplicationFilters.Program
-        Service         = $properties.ServiceFilters.Service
+    $Result = @{
+        Ensure = 'Present'
+    }
+
+    # Populate the properties for get target resource by looping through
+    # the parameter array list and adding the values to 
+    foreach ($p in $ParameterList)
+    {
+
+        if ($p.type -eq 'Array')
```

Comment:
Is the real difference in code here the verbose message?

If so I'm thinking one of two things.

1) Cut it, I mean it's nice to know don't get me wrong but seems strange to fork the code here just for a better formatted verbose message.
2) Move the $Value and $Result out of the if/else bracket and just have it run the different verbose message. This reduces the duplicated code and maintains the verbose message.  Also clarifies where the fork in the logic is actually occuring.